src/gotcha_dl.c

```diff
@@ -56,12 +186,12 @@ static void* dlsym_wrapper(void* handle, const char* symbol_name){
   debug_printf(1, "User called dlsym(%p, %s)\n", handle, symbol_name);
 
   if(handle == RTLD_NEXT){
-    return _dl_sym(RTLD_NEXT, symbol_name ,__builtin_return_address(0));
+    return gotchas_dlsym_rtld_next_lookup(RTLD_NEXT, symbol_name ,__builtin_return_address(0));
```

Comment:
As mentioned above, I suggest having gotchas_dlsym_rtld_next_lookup() return a link_map. We can then use that link_map to make a call like:
```
return orig_dlsym(link_map, symbol_name)
```

That lets GLIBC's dlsym() do the actual symbol resolution, but limits it to the library we identified as the proper RTLD_NEXT.  As mentioned, symbol resolution has a whole bunch of corner cases we don't want to handle.

Reply:
fixed. Returning the lmap. and then letting original function do the symbol lookup on lmap.